release/models/interfaces/openconfig-interfaces.yang

```diff
@@ -197,6 +204,16 @@ module openconfig-interfaces {
       name and subinterface index.";
   }
 
+  typedef interface-full-id {
+    type string {
+      pattern '(.+)\.([0-9]+)';
```

Comment:
I'm not sure we can impose a restriction via pattern here unless we always can assume a subinterface is in existance - also see https://github.com/openconfig/public/issues/781 on the issue w/ tunnel/routed-vlan interfaces

A description statement may be the only way to cover that this is meant to either be an exact correlation to an interface name (should no subinterfaces exist) or `interface.subinterface`

If we make the regex cover anything post the `.` is optional then there is no use for the regex.

I'd vote for covering this in a very descriptive description stmt rather